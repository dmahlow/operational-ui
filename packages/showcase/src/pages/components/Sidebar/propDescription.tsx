/* eslint max-len: 0 */
import * as React from "react"

export default {
  sidebarItem: [
    {
      name: "label",
      description: "The label of the SidebarItem.",
      defaultValue: "",
      type: "string",
      optional: false
    },
    {
      name: "open",
      description: "Is the item open or closed by default?",
      defaultValue: "false",
      type: "boolean",
      optional: false
    },
    {
      name: "onClick",
      description:
        "A function to pass to the item that executes before the item expands. If a function returning a Promise is passed in, the item only expands after the Promise resolves.",
      defaultValue: <pre>() => this.open = !this.open</pre>,
      type: "func",
      optional: true
    }
  ],
  sidebarLink: [
    {
      name: "to",
      description: "Created to work with react-router, this wraps the children in a <Link> to your route.",
      defaultValue: "",
      type: "string",
      optional: true
    },
    {
      name: "onClick",
      description:
        "A function called on click of this component, to be used instead of the `to` prop to do more than just navigate.",
      defaultValue: "",
      type: "func",
      optional: true
    },
    {
      name: "symbol",
      description:
        "A symbol to display at the right-hand side of the link, such as a `%` sign to suggest a unit of measure.",
      defaultValue: "",
      type: "string",
      optional: true
    },
    {
      name: "color",
      description:
        "Different links can have different colors to communicate different use-cases or purposes. This can be a hex value, or a named color in your theme.",
      defaultValue: "The primary color of your theme.",
      type: "string",
      optional: true
    }
  ]
}
