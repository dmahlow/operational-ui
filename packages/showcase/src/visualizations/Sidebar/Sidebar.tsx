import * as React from "react"
import { Sidebar, SidebarItem, SidebarLink } from "contiamo-ui-components"

type Props = {
  location?: {
    pathname: string
  }
  css: {}
}

const paths: {
  [key: string]: Array<string>
} = {
  processFlow: ["/case01"],
  barChart: [""],
}

export default ({ location, css }: Props) => (
  <Sidebar css={css}>
    <SidebarItem open={location && paths.processFlow.includes(location.pathname)} label="Process Flow">
      <SidebarLink to="/visualizations/processFlow/case01">Case 1</SidebarLink>
    </SidebarItem>
    <SidebarItem open={location && paths.barChart.includes(location.pathname)} label="Bar chart" />
  </Sidebar>
)
