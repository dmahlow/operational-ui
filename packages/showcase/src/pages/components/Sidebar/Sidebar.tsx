import * as React from "react"
import { Link } from "react-router-dom"

import Playground from "component-playground"
import { Sidebar as DemoSidebar, SidebarItem, SidebarLink, CardHeader } from "contiamo-ui-components"

import Table from "../../../components/PropsTable/PropsTable"
import snippet from "./snippets/sidebar.snippet"
import propDescription from "./propDescription"
import { toReactPlayground } from "../../../utils/snippet"

export const fetch = (text: any) => new Promise(resolve => setTimeout(() => resolve(text), 2000))

export default () =>
  <div>
    <CardHeader>Sidebar</CardHeader>

    <p>
      The sidebar is a dynamic list-style navigational element to be used in cases with a large number of list-style
      actionable items. This component involves composition of two constituent elements. Namely,
    </p>
    <ul>
      <li>
        <a href="#sidebar-item">SidebarItem</a>
      </li>
      <li>
        <a href="#sidebar-link">SidebarLink</a>
      </li>
    </ul>

    <div style={{ marginBottom: 32 }} />

    <h4>Usage</h4>
    <Playground
      codeText={toReactPlayground(String(snippet))}
      scope={{ React, Sidebar: DemoSidebar, SidebarItem, SidebarLink, fetch }}
    />

    <div style={{ marginBottom: 32 }} />

    <CardHeader id="sidebar-item">SidebarItem</CardHeader>
    <h4>An expandable group of SidebarLinks, with added asynchronous functionality.</h4>

    <Table props={propDescription.sidebarItem} />
    <p style={{ marginTop: 16, marginBottom: 32 }}>
      <strong>
        Note: This component is wrapped with <Link to="/tooltips">withTooltip</Link> and thus exposes all of the props
        that such components do.
      </strong>
    </p>

    <CardHeader id="sidebar-link">SidebarLink</CardHeader>
    <p>A link, but with onClick instead of href.</p>

    <Table props={propDescription.sidebarLink} />
    <p style={{ marginTop: 16, marginBottom: 32 }}>
      <strong>
        Note: This component is wrapped with <Link to="/tooltips">withTooltip</Link> and thus exposes all of the props
        that such components do.
      </strong>
    </p>
  </div>
