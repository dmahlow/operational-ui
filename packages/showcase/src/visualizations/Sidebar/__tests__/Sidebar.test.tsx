import * as React from "react"
import { shallow } from "enzyme"

import Sidebar from "../Sidebar"

describe("App Showcase: Sidebar", () => {
  it("Should render correctly", () => {
    expect(shallow(<Sidebar />)).toMatchSnapshot()
  })
})
