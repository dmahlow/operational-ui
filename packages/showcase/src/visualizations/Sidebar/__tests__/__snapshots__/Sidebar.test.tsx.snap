// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Showcase: Sidebar Should render correctly 1`] = `
<glamorous(Sidebar)>
  <glamorous(SidebarItem)
    label="Process Flow"
  >
    <glamorous(SidebarLink)
      to="/visualizations/processFlow/case01"
    >
      Case 1
    </glamorous(SidebarLink)>
  </glamorous(SidebarItem)>
  <glamorous(SidebarItem)
    label="Bar chart"
  />
</glamorous(Sidebar)>
`;
